ui/app/abilities/variable.js

```diff
@@ -116,22 +116,29 @@ export default class Variable extends AbstractAbility {
 
   @computed('allPaths', 'namespace', 'path', 'token.selfTokenPolicies')
   get policiesSupportVariableWriting() {
-    if (this.namespace === WILDCARD_GLOB && this.path === WILDCARD_GLOB) {
-      // If you're checking if you can write from root, and you don't specify a namespace,
-      // Then if you can write in ANY path in ANY namespace, you can get to /new.
+    if (this.path === WILDCARD_GLOB) {
+      // If checking for write permission on the root path
       return this.policyNamespacesIncludeVariablesCapabilities(
         this.token.selfTokenPolicies,
         ['write'],
-        this._nearestMatchingPath(this.path)
+        WILDCARD_GLOB
       );
     } else {
-      // Checking a specific path in a specific namespace.
-      // TODO: This doesn't cover the case when you're checking for the * namespace at a specific path.
-      // Right now we require you to specify yournamespace to enable the button.
+      // Checking a specific path
       const matchingPath = this._nearestMatchingPath(this.path);
-      return this.allPaths
-        .find((path) => path.name === matchingPath)
-        ?.capabilities?.includes('write');
+      if (this.namespace === WILDCARD_GLOB) {
+        // Checking for the * namespace at a specific path
+        return this.policyNamespacesIncludeVariablesCapabilities(
+          this.token.selfTokenPolicies,
+          ['write'],
+          matchingPath
+        );
```

Comment:
For clarity and because I find `policyNamespacesIncludeVariablesCapabilities` a little hard to follow... does this check that the user has write-access to this path in _any_ namespace, or _all_ namespaces?

Reply:
This will check for the presence of the request capabilities in at least one of the user's namespaces (at the specified path)

Reply:
I'm going to change up the internal var name `namespacesWithVariableCapabilities` to be something like `variableCapabilitiesAmongNamespaces` or something to make it clearer